Azure-Samples/azure-ml-federated-learning
# Fix optional input yaml and mlflow log bugs

## Purpose

* Incorporate recent changes related to optional inputs syntax change in DPv2.
* Fix mlflow logging issue- As we were using auto logging that requires restructuring PyTorch code to PyTorch-lightning. Hence, switching to manual logging.

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code


Submit an example pipeline and go to any silo job metrics to see train/test loss.
